src/Application.vue

```diff
@@ -144,9 +144,10 @@ export default {
       const widgetData = widgetInstanceData || Fliplet.Widget.getData(widgetId) || {}
 
       handleWidgetData(widgetData)
```

Comment:
Is this function used anywhere else in the code?
If there is, have you tested that flow to make sure there are no issues? Use a slow connection to identify race conditions easily.

Reply:
I'm asking because this function wasn't a Promise, and now it is, so if we use data that it sets after the function is run, you might need to restructure the flow to account for a possible wait/request.

Reply:
Yes, it is used at this line https://github.com/YaroslavOvdii/fliplet-widget-theme/blob/37c4464cd0589f4de354a92208963cba37d0a41d/src/Application.vue#L515 but as I can see we do not need a result from it so I didn't change this part.